### DISCUSSION TITLE
Support for Resolving Indirect Function Pointers to Direct Calls in Decompiled View

### DISCUSSION THREAD

**Discussion by gtnick19** [NationalSecurityAgency/ghidra#7188] [Q&A]
This was the initial problem I had: 

In Ghidra, I'm working with functions that use indirect pointers (e.g., single or double indirect calls) to call other functions, often represented by `BLX ` instructions in ARM assembly. I need a way to automatically replace these indirect function pointers in the decompiled view with the actual function names, showing direct calls (e.g., `disable_interrupts()` instead of `(*(code *)PTR_disable_interrupts))`.

Is there a way to force Ghidra’s decompiler to fully dereference these function pointers and display the resolved function names in place of the indirect calls? Despite adding direct references and flow overrides, I haven't found a way for the decompiler to reflect these changes, and manual re-decompilation attempts haven’t resolved the issue.

I was able to fix it by doing this: Set pointer settings of the variable to have a mutability of constant.

New Question: Does anyone have a script or know to script it so that I can have this occur throughout my whole program and resolve all occurrences at once ?

**Comment by astrelsky** [comment]
You can change the writeable bit of the memory block if you have something like the .plt section that is marked as writeable for whatever reason.
